core/node/eth_watch/src/event_processors/appended_chain_batch_root.rs

```diff
@@ -0,0 +1,236 @@
+use std::sync::Arc;
+
+use anyhow::Context;
+use itertools::Itertools;
+use zksync_dal::{eth_watcher_dal::EventType, Connection, Core, CoreDal, DalError};
+use zksync_mini_merkle_tree::MiniMerkleTree;
+use zksync_types::{
+    api::{ChainAggProof, Log},
+    ethabi, h256_to_u256,
+    l2_to_l1_log::{
+        BatchAndChainMerklePath, BATCH_LEAF_PADDING, LOG_PROOF_SUPPORTED_METADATA_VERSION,
+    },
+    u256_to_h256, L1BatchNumber, L2ChainId, SLChainId, H256, U256,
+};
+
+use crate::{
+    client::L2EthClient,
+    event_processors::{EventProcessor, EventProcessorError, EventsSource},
+};
+
+/// Responsible for `AppendedChainBatchRoot` events and saving `BatchAndChainMerklePath` for batches.
```

Comment:
Could you write a short introduction what this type of event represents and how it's used?